adam/visualization/make_scenes.py

```diff
@@ -644,33 +643,45 @@ def objects_to_freeze(
         return immutableset([])
     frozen_objects = []
     for action in semantics_situation.actions:
-        if action.action_type == GO:
-            agent: Optional[SituationObject] = None
-            goal: Optional[Union[SituationObject, Region[SituationObject]]] = None
-            for (
-                ontology_node,
-                object_or_region,
-            ) in action.argument_roles_to_fillers.items():
-
-                if ontology_node.handle == "goal":
-                    goal = object_or_region
-                elif ontology_node.handle == "agent":
-                    agent = object_or_region
-
-            if (
-                agent
-                and goal
-                and isinstance(agent, SituationObject)
-                and isinstance(goal, Region)
-            ):
-                for region_relation in in_region_map[situation_obj_to_handle[agent]]:
-                    if (
-                        region_relation.reference_object.debug_handle
-                        == situation_obj_to_handle[goal.reference_object]
-                    ):
-                        frozen_objects.append(
-                            situation_obj_to_handle[goal.reference_object]
-                        )
+        agent: Optional[SituationObject] = None
+        theme: Optional[SituationObject] = None
+        goal: Optional[Union[SituationObject, Region[SituationObject]]] = None
+        for (ontology_node, object_or_region) in action.argument_roles_to_fillers.items():
+
+            if ontology_node.handle == "goal":
```

Comment:
I think you should just import GOAL, AGENT, and THEME to compare with rather than using the debug handles

Reply:
addressed in latest commit.